python/ray/job_config.py

```diff
@@ -67,6 +54,24 @@ def serialize(self):
         job_config = self.get_proto_job_config()
         return job_config.SerializeToString()
 
+    def set_runtime_env(self, runtime_env):
+        # Lazily import this to avoid circular dependencies.
+        import ray._private.runtime_env as runtime_support
+        if runtime_env:
+            # Remove working_dir from the dict here, since that needs to be
+            # uploaded to the GCS after the job starts.
+            without_dir = dict(runtime_env)
+            if "working_dir" in without_dir:
+                del without_dir["working_dir"]
+            self._parsed_runtime_env = runtime_support.RuntimeEnvDict(
+                without_dir)
+            self.worker_env = self._parsed_runtime_env.to_worker_env_vars(
+                self.worker_env)
+        else:
+            self._parsed_runtime_env = runtime_support.RuntimeEnvDict({})
+        self.runtime_env = runtime_env or dict()
+        self._cached_pb = None
```

Comment:
why is this defined here?

Reply:
We can remove it from the calling function I guess. But is either of these places correct?

Reply:
We definitely need this one, otherwise we won't re-serialize the protobuf. I guess we can remove the other one now though.